railties/lib/rails/application.rb

```diff
@@ -590,5 +590,46 @@ def build_request(env)
       def build_middleware
         config.app_middleware + super
       end
+
+      class NonSymbolAccessDeprecatedHash < Hash # :nodoc:
```

Comment:
Would it make more sense to call this `StringAccessDeprecatedHash` and check for string access or are we genuinely concerned with all cases that aren't symbols? String access is all we are testing for.

Reply:
Since the new behaviour restricts usage to symbol keys only, I wanted to make sure that we generate a deprecation notice for anything that is not a symbol. For all intents and purposes, you are correct, we are really trying to detect string access, but I thought it wouldn't hurt to be more general in this case.

I am open to changing the behaviour and renaming the class, if you think otherwise, though.

Reply:
> I am open to changing the behaviour and renaming the class, if you think otherwise, though.

I don't feel strongly about it. Let's get a second opinion.

Reply:
I'm fine with the name and implementation. We can only have strings and symbols keys anyway.